src/NuGet.Clients/NuGet.PackageManagement.UI/Common/ErrorFloodGate.cs

```diff
@@ -20,9 +20,9 @@ internal class ErrorFloodGate
         private readonly ConcurrentQueue<int> _failures = new ConcurrentQueue<int>();
 
         private DateTimeOffset _lastEvaluate = DateTimeOffset.Now;
-        private bool _isOpen = false;
+        private bool _hasTooManyNetworkErrors = false;
 
-        public bool IsOpen
+        public bool HasTooManyNetworkErrors
```

Comment:
This is effectively implementing the circuit-breaker pattern, but from what I can see, once it triggers, only a process restart will reset it, even if it's just a transient network error (common on wifi).

Should we consider using [Polly](https://www.nuget.org/packages/Polly/), to have a well-tested implementation that has more features without having to implement it ourselves?

edit: I just saw the comment where this is being used, which explains why it's here. Maybe my suggestion to reset the circuit breaker doesn't make sense in this scenario, although I haven't thought about this scenario for long enough to determine for myself if it makes sense.

Reply:
@alpaix implemented the errorfloodgate.cs several years ago. I'd prefer to minimize changes...and focus on the async improvements. unless we have strong reasons to change.
(i think my naming makes that type more approachable...)